rad/src/reducers/median.rs

```diff
@@ -0,0 +1,318 @@
+            } else {
+                // Even number of elements: take average of element at (N/2 - 1) and N/2
+                let right_pos = input_not_nan.len() / 2;
+                let right_elem = input_not_nan[right_pos].into_inner();
+                let left_pos = right_pos - 1;
+                let left_elem = input_not_nan[left_pos].into_inner();
+
+                // Create new array to be able to use average::mean reducer
+                let rl = RadonArray::from(vec![
+                    RadonTypes::Float(RadonFloat::from(left_elem)),
+                    RadonTypes::Float(RadonFloat::from(right_elem)),
+                ]);
+                // MeanReturnPolicy only applies to integers, so this will actually return a float
+                mean(&rl, MeanReturnPolicy::RoundToInteger)
+            }
+        }
+        Some(RadonTypes::Integer(_)) => {
+            let sorted_input =
+                match array_operators::sort(input, &[], &mut ReportContext::default()) {
+                    Ok(RadonTypes::Array(arr)) => arr.value(),
+                    Ok(_different_type) => unreachable!(),
+                    Err(e) => return Err(e),
+                };
+
+            if sorted_input.is_empty() {
+                // This is unreachable
+                Err(RadError::ModeEmpty)
+            } else if sorted_input.len() % 2 == 1 {
+                // Odd number of elements: take element at floor(N/2):
+                let median_pos = sorted_input.len() / 2;
+
+                Ok(sorted_input[median_pos].clone())
+            } else {
+                // Even number of elements: take average of element at (N/2 - 1) and N/2
+                let right_pos = sorted_input.len() / 2;
+                let left_pos = right_pos - 1;
+
+                // Create new array to be able to use average::mean reducer
+                let rl = RadonArray::from(vec![
+                    sorted_input[left_pos].clone(),
+                    sorted_input[right_pos].clone(),
+                ]);
+                // RoundToInteger means that when the average is not an integer, it will be rounded to an
+                // integer. For example, the average of 1 and 2, which is 1.5, will be rounded to 1.
+                mean(&rl, MeanReturnPolicy::RoundToInteger)
+            }
+        }
+        Some(_rad_types) => Err(RadError::UnsupportedReducer {
+            array: input.clone(),
+            reducer: RadonReducers::AverageMedian.to_string(),
+        }),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::{
+        current_active_wips,
+        operators::array::reduce,
+        types::{float::RadonFloat, integer::RadonInteger, string::RadonString},
+        RadError::ModeEmpty,
+    };
+    use serde_cbor::Value;
+
+    #[test]
+    fn test_operate_reduce_median_empty() {
+        let input = RadonArray::from(vec![]);
+        let output = median(&input).unwrap_err();
+        let expected_error = ModeEmpty;
```

Comment:
TODO: we use `ModeEmpty` as the error for when the median reducer is applied over an empty array. Should we create a new error `MedianEmpty`, or rename `ModeEmpty` to a more generic `EmptyArray` error?

Reply:
I think EmptyArray sounds better

Reply:
Let's take this into account for the eventual WIP, sirs.

Reply:
So it looks like this error cannot be easily renamed because it appears in the `UnhandledIntercept` error message, so changing its name would make some tallies invalid. However we could maybe avoid the `UnhandledIntercept` by implementing the conversion from `RadError` to `RadonError` and assigning an error code to this error. The currently assigned error codes can be found [here in the bridge](https://github.com/witnet/witnet-ethereum-bridge/blob/b18af3ca43454316f6034d5e131691344687db87/contracts/libs/Witnet.sol#L69) and [here in the node](https://github.com/witnet/witnet-rust/blob/44522fc1d5253da2dcd8d42a3d614aa60ac1772e/data_structures/src/radon_error.rs#L14).

Reply:
I could add a TODO or FIXME and create a new issue, because I think it is out of the scope of this issue. What do you think?

Reply:
We can use EmptyArray only in the median by the moment